server/bin/service-key.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""Create an imaginary service account with a single API key.
+
+Create a User record and a matching API key in the database, which can be used
+as a service account. The API key provides authentication for services and the
+User record allows for ID to name translation.
+
+The primary purpose is to serve as a "legacy" collector for the 0.69-11
+passthrough server, but it could also be used for other API clients which don't
+require a specific SSO user identity.
+"""
+
+from argparse import ArgumentParser
+from datetime import datetime, timezone
+import os
+from pathlib import Path
+import sys
+import uuid
+
+import jwt
+
+from pbench.common.logger import get_pbench_logger
+from pbench.server import PbenchServerConfig
+from pbench.server.database import init_db
+from pbench.server.database.models.api_keys import APIKey
+from pbench.server.database.models.users import User
+
+
+def main(options, name):
+    """Create a Pbench Server service account"""
+    config = PbenchServerConfig.create(options.cfg_name)
+    logger = get_pbench_logger(name, config)
+    init_db(config, logger)
+
+    username = options.username
+
+    # You can specify a nonsense string or let the code generate a real (and
+    # "very likely unique") UUID.
+    id = options.identity if options.identity else str(uuid.uuid1())
+
+    if User.query(id=id):
+        print(f"The user ID {id!r} already exists", file=sys.stderr)
+        return 1
+    if User.query(username=username):
+        print(f"The username {username!r} already exists", file=sys.stderr)
+        return 1
+
+    # Create a user proxy object, but don't add it to the session yet.
+    user = User(id=id, username=username)
+
+    # NOTE: this needs to closely mimic ApiKey.generate_api_key, however we
+    # can't call that here as it depends on a Flask app context. Luckily, the
```

Comment:
It seems like we have several alternatives to replicating the code.
- The best alternative (probably) is to do a modest refactoring of `ApiKey.generate_api_key()` so that the dependencies (which I think are just the secret and the logger) are passed in.  That should make it straightforward to share the code.
- We could consider using Flask in this script.  I think it's possible to do so with almost no configuration, so there would be little impact (other than code bloat).
- We could consider faking/mocking/patching the Flask module/import (this is Python, after all!  😉).  (Although, that cure might be worse than the disease, but I don't know -- I really hate code duplication, especially in modules which aren't tested.)